src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3743,12 +3744,42 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                     if (!memberSignatures.TryGetValue(targetProperty, out var existingMember))
                     {
                         existingMember = OverriddenOrHiddenMembersHelpers.FindFirstHiddenMemberIfAny(targetProperty, memberIsFromSomeCompilation: true);
-                        isInherited = true;
+                        if (existingMember is FieldSymbol)
+                        {
+                            // field from base type should not be considered before we look at fields from current type
+                            existingMember = null;
+                        }
+                        else
+                        {
+                            isInherited = true;
```

Comment:
I didn't understand why we avoid setting this for fields. In this case, the field is also inherited, right? We don't use it, but at the same time it seems simpler if we set it unconditionally.

Reply:
We shouldn't set `isInherited` to `true` when a field was found from base types here, because we may find a field on the current type (which would be preferred and is not inherited).